pymeasure/instruments/oxfordinstruments/ips120_10.py

```diff
@@ -110,28 +108,29 @@ class IPS120_10(Instrument):
         8: "No switch fitted",  # No switch fitted
     }
 
-    def __init__(self, resourceName,
+    def __init__(self,
+                 adapter,
+                 name="Oxford IPS",
                  clear_buffer=True,
                  switch_heater_heating_delay=None,
                  switch_heater_cooling_delay=None,
                  field_range=None,
                  **kwargs):
 
+        kwargs.setdefault('read_termination', '\r')
         super().__init__(
-            resourceName,
-            "Oxford IPS",
+            adapter=adapter,
+            name=name,
             includeSCPI=False,
-            send_end=True,
-            read_termination="\r",
-            **kwargs
+            gpib={'send_end': True},
+            asrl={'baud_rate': 9600,
+                  'data_bits': 8,
+                  'parity': 0,
+                  'stop_bits': 20,
+                  },
+            **kwargs,
         )
```

Comment:
At this moment, this adapter is not used, though it is still required for these instruments; @bilderbuchi, how would you recommend to use such a custom VISAAdapter with the new default connection parameters?

Edit:
Would something like this be the best option?
``` python
    def __init__(self,
                 adapter,
                 name="Oxford IPS",
                 clear_buffer=True,
                 switch_heater_heating_delay=None,
                 switch_heater_cooling_delay=None,
                 field_range=None,
                 **kwargs):


        if isinstance(adapter, (int, str)):
            kwargs.setdefault('read_termination', '\r')
            adapter = VISAAdapter(
                adapter, 
                gpib={'send_end': True},
                asrl={'baud_rate': 9600,
                      'data_bits': 8,
                      'parity': 0,
                      'stop_bits': 20,
                },
                **kwargs,
            )


        super().__init__(
            adapter=adapter,
            name=name,
            includeSCPI=False,
        )
```

Reply:
Yes, this looks reasonable. The `VISAAdapter` has to be `RetryingVISAAdapter_oxfordinstruments` (a better name might be `OxfordInstrumentsAdapter`, if you agree)
As you say this adapter type is mandatory, I would add an else clause with a check to confirm that `adapter` (i.e. if not int or string) is of the correct type.

Reply:
Thanks.

I agree I the naming. I think I won't add an else clause, since the type of adapter is not that mandatory; if someone wants to use a normal Visa or serial adapter, it will also work, but you run the risk of it being at more prone to errors.